docs/build/tutorials/nodes-and-staking/set-up-an-avalanche-node-with-google-cloud-platform.md

```diff
@@ -0,0 +1,139 @@
+
+**NOTE**
+
+Depending upon how you intend to execute your terraform operations you may or may not need to enable public access to the bucket. Obviously, not exposing the bucket for `public` access (even if authenticated) is preferable. If you intend to simply run terraform commands from your local machine then you will need to open the access up.
+I recommend to employ a full CI/CD pipeline using GCP Cloud Build which if utilized will mean the bucket can be marked as `private`. A full walkthrough of Cloud Build setup in this context can be found [here](https://cloud.google.com/architecture/managing-infrastructure-as-code)
+
+### Clone Github Repository
+
+I have provided a rudimentary terraform construct to provision a node on which to run Avalanche which can be found [here](https://github.com/null-ref-ex/avax-on-gcp/tree/main/terraform).
+Documentation below assumes you are using this repository but if you have another terraform skelaton similar steps will apply.
+
+### Terraform Configuration
+
+1. If running terraform locally, please [install](https://learn.hashicorp.com/tutorials/terraform/install-cli) it.
+2. In this repository, navigate to the `terraform` directory.
+3. Under the `projects` directory, rename the `my-avax-project` directory to match your GCP project name that you created (not required, but nice to be consistent)
+4. Under the folder you just renamed locate the `terraform.tfvars` file.
+5. Edit this file and populate it with the values which make sense for your context and save it.
+6. Locate the `backend.tf` file in the same directory.
+7. Edit this file ensuring to replace the `bucket` property with the GCS bucket name that you created earlier.
+
+If you do not with to use cloud storage to persist terraform state then simply switch the `backend` to some other desirable provider.
+
+### Terraform Execution
+
+Terraform enables us to see what it would do if we were to run it without actually applying any changes... this is called a `plan` operation. This plan is then enacted (optionally) by an `apply`.
+
+#### Plan
+
+1. In a terminal which is able to execute the `tf` binary, `cd` to the ~`my-avax-project` directory that you renamed in step 3 of `Terraform Configuration`.
+2. Execute the command `tf plan`
+3. You should see a JSON output to the stdout of the terminal which lays out the operations that terraform will execute to apply the intended state.
+
+#### Apply
+
+1. In a terminal which is able to execute the `tf` binary, `cd` to the ~`my-avax-project` directory that you renamed in step 3 of `Terraform Configuration`.
+2. Execute the command `tf apply`
+
+If you want to ensure that terraform does **exactly** what you saw in the `apply` output, you can optionally request for the the `plan` output to be saved to a file to feed to `apply`. This is generally considered best practice in highly fluid environments where rapid change is occuring from multiple sources.
```

Comment:
Could you add a short Summary section at the end? It reads better if there is a conclusion to the article.

Reply:
Will do.